src/MyHealth/SelectSymptoms.tsx

```diff
@@ -29,13 +29,22 @@ import {
 } from "../styles"
 import { SvgXml } from "react-native-svg"
 import { Icons } from "../assets"
+import { MyHealthStackParams } from "../navigation/MyHealthStack"
 
 const SelectSymptomsScreen: FunctionComponent = () => {
   useStatusBarEffect("dark-content", Colors.primaryLightBackground)
   const { t } = useTranslation()
   const navigation = useNavigation()
-  const [selectedSymptoms, setSelectedSymptoms] = useState<string[]>([])
-  const { addLogEntry } = useSymptomLogContext()
+  const route = useRoute<RouteProp<MyHealthStackParams, "SelectSymptoms">>()
+  const { addLogEntry, updateLogEntry } = useSymptomLogContext()
+
+  const { logEntry } = route.params
+  const symptomLogEntryToEdit = logEntry ? JSON.parse(logEntry) : null
```

Comment:
There's a few nulls floating around in this component now. I wonder if it makes sense to separate the concepts of `new` and `edit` a bit here? for example should we have a two components, `CreateSymptomsLogForm` and `EditSymptomsForm` ?

or maybe the form want so accept a prop of `symptoms: Symptom[]` or something and it defaults to an empty list?

Not 100% sure, but feels like finding a way to pull the nulls up the flow a bit will lead to less complexity in the form.

Reply:
How do you feel about handling that in a refactor once this and the delete functionality are merged?